src/TufValidatedComposerRepository.php

```diff
@@ -88,9 +95,59 @@ protected function configurePackageTransportOptions(PackageInterface $package)
             'repository' => $config['url'],
             'target' => $package->getName() . '/' . $package->getVersion(),
         ];
+        if ($this->updater) {
+            $options['max_file_size'] = $this->updater->getLength($options['tuf']['target']);
+        }
         $package->setTransportOptions($options);
     }
 
+    /**
+     * Extracts a TUF target path from a full URL.
+     *
+     * @param string $url
+     *   A URL.
+     *
+     * @return string
+     *   A TUF target path derived from the URL.
+     */
+    private function getTargetFromUrl(string $url): string
+    {
+        $config = $this->getRepoConfig();
+        $target = str_replace($config['url'], null, $url);
+        return ltrim($target, '/');
+    }
+
+    /**
+     * Reacts before metadata is downloaded.
+     *
+     * @param PreFileDownloadEvent $event
+     *   The event object.
+     */
+    public function prepareMetadata(PreFileDownloadEvent $event): void
+    {
+        if ($this->updater) {
+            $target = $this->getTargetFromUrl($event->getProcessedUrl());
+            $options = $event->getTransportOptions();
+            try {
+                $options['max_file_size'] = $this->updater->getLength($target);
+            } catch (NotFoundException $e) {
+                // As it compiles information on the available packages, ComposerRepository
+                // expects to receive occasional 404 responses from the server, which
+                // it treats as a totally normal indication that the repository doesn't have
+                // a particular package, or version of a package. Those requests probably
```

Comment:
So is the only reason we need to do this is for 404 responses.  Do they have a body length? If they do I imagine they would very small correct? Could we have our own constant `MAX_404_LENGTH` or something return 
`static::MAX_404_LENGTH < static::MAXIMUM_DOWNLOAD_BYTES ? static::MAX_404_LENGTH : static::MAXIMUM_DOWNLOAD_BYTES`

Even though `static::MAXIMUM_DOWNLOAD_BYTES` is small minimize as much as we can how much we are willing to download if a target is not known.

Reply:
Makes sense. I constrained it to 1024 bytes.